code/tier_classification/tier_classification.R

```diff
@@ -0,0 +1,247 @@
+  # filter based on OncoKB annotation level 
+  dplyr::filter(HIGHEST_LEVEL %in% c("LEVEL_4","LEVEL_R2") | HIGHEST_DX_LEVEL %in% c("LEVEL_Dx3") | HIGHEST_PX_LEVEL %in% c("LEVEL_Px3")) %>%
+  # contain only variants that are <1% (less likely to be germline)
+  dplyr::filter(gnomAD_AF<0.01| is.na(gnomAD_AF)) %>%
+  # filter on VAF over 5% to avoid background noise
+  dplyr::filter(VAF>0.05) %>%
+  # filter to contain only mutations that has COSMIC ID
+  dplyr::mutate(has_COSMIC = dplyr::case_when(
+    stringr::str_detect(Existing_variation,
+                        "COSM") ~ "Yes",
+    TRUE ~ "No"
+  )) %>%
+  dplyr::filter(has_COSMIC == "Yes") %>%
+  # filter to contain only mutations with citations
+  dplyr::filter(!is.na(CITATIONS)) %>% 
+  # filter to TSG lists in OMPARE database
+  dplyr::filter(Hugo_Symbol %in% oncogenes_tsg ) %>% 
+  dplyr::select(-has_COSMIC) %>% 
+  dplyr::mutate(tier_classification = "TIER: II")
+
+
+# Filter for Tier III
+oncokb_anno_tier3 <- oncokb_anno %>% 
+  # filter to variation of interest 
+  dplyr::filter(Variant_Classification %in% c('Missense_Mutation',  # filter the files 
+                                              'In_Frame_Del', 'In_Frame_Ins')) %>%
+  # filter to samples that do not have OncoKB annotation 
+  dplyr::filter(is.na(HIGHEST_LEVEL)) %>%
+  dplyr::filter(is.na(HIGHEST_DX_LEVEL)) %>%
+  dplyr::filter(is.na(HIGHEST_PX_LEVEL)) %>%
+  # contain only variants that are <1% (less likely to be germline)
+  dplyr::filter(gnomAD_AF<0.01| is.na(gnomAD_AF)) %>%
+  # filter on VAF over 5% to avoid background noise
+  dplyr::filter(VAF>0.05) %>%
+  dplyr::mutate(tier_classification = "TIER: III")
+
+# Filter for Tier IV
+oncokb_anno_tier4 <- oncokb_anno %>% 
+  # filter to variation of interest 
+  dplyr::filter(Variant_Classification =='Missense_Mutation') %>%
+  # filter to samples that do not have OncoKB annotation or related citations
+  dplyr::filter(is.na(HIGHEST_LEVEL)) %>%
+  dplyr::filter(is.na(HIGHEST_DX_LEVEL)) %>%
+  dplyr::filter(is.na(HIGHEST_PX_LEVEL)) %>%
+  dplyr::filter(is.na(CITATIONS)) %>% 
+  # contain only variants that are <1% (less likely to be germline)
+  dplyr::filter(gnomAD_AF>=0.01) %>%
+  # filter on VAF over 5% to avoid background noise
+  dplyr::filter((VAF>=0.4 & VAF<=0.6) | VAF>=0.9) %>%
+  dplyr::filter(!grepl("deleterious", SIFT)) %>%
+  dplyr::filter(!grepl("damaging", PolyPhen)) %>%
+  dplyr::mutate(tier_classification = "TIER: IV")
+
+# combine all the variants in MAF that contain Tier information 
+oncokb_anno_tiers <- bind_rows(oncokb_anno_tier1, oncokb_anno_tier2, oncokb_anno_tier3, oncokb_anno_tier4) %>%
+  dplyr::select(Hugo_Symbol, Variant_Classification, HGVSp_Short, tier_classification)
+
+
+######################### annotate MSK SNV hotspot database to key clinical findings & all findings table
+
+# annotate SNV first
+all_findings_output_hotspots <- data.frame()
+key_clinical_findings_output_hotspots <- data.frame()
+
+for(p in 1:nrow(hotspot_snv)){
```

Comment:
I wonder if this could have been accomplished via a join between data tables in lieu of a for loop (e.g. matching on gene and position, then annotating variant_properties for genes that intersected between the two tables). Just a thought though not necessary to change if performance isn't an issue.

Reply:
I thought about that as well but because the `Details` column is a comma-separated string, it would be easier to use `grepl`

Reply:
Let me quickly update this!